src/ucs/memory/rcache.c

```diff
@@ -424,7 +426,10 @@ static void ucs_mem_region_destroy_internal(ucs_rcache_t *rcache,
     --rcache->num_regions;
     region_size         = region->super.end - region->super.start;
     rcache->total_size -= region_size;
-    --rcache->distribution[ucs_rcache_distribution_get_bin(region_size)];
+
+    distribution_bin = ucs_rcache_distribution_get_bin(region_size);
```

Comment:
maybe ucs_rcache_distribution_get_bin could get (rcache, region_size) and return pointer to the distribution struct?